tests/test_10_use_cases.py

```diff
@@ -0,0 +1,210 @@
+            - Query the L2VPN and check the status to validate it is back UP
+            - Run a PING test between hosts to make sure we have connectivity
+        '''
+        api_url = SDX_CONTROLLER + '/l2vpn/1.0'
+        payload = {
+            "name": "Test L2VPN creation",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet01:50","vlan": "100"},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "100"}
+            ]
+        }
+        response = requests.post(api_url, json=payload)
+        assert response.status_code == 201, response.text
+
+        # wait until status changes for under provisioning to up
+        time.sleep(5)
+
+        data = requests.get(api_url).json()
+        key = next(iter(data))
+        assert data[key]["status"] == "up"
+
+        h6, h8 = self.net.net.get('h6', 'h8')
+        h6.cmd('ip link add link %s name vlan100 type vlan id 100' % (h6.intfNames()[0]))
+        h6.cmd('ip link set up vlan100')
+        h6.cmd('ip addr add 10.1.1.6/24 dev vlan100')
+        h8.cmd('ip link add link %s name vlan100 type vlan id 100' % (h8.intfNames()[0]))
+        h8.cmd('ip link set up vlan100')
+        h8.cmd('ip addr add 10.1.1.8/24 dev vlan100')
+
+        # test connectivity
+        assert ', 0% packet loss,' in h6.cmd('ping -c4 10.1.1.8')
+
+        # set one link to down
+        self.net.net.configLinkStatus('Tenet01', 'Tenet03', 'down')
+
+        time.sleep(15)
+
+        '''
+            HERE: The following code verifies that the link was set to down, 
+            and that the current_path is Tenet01-Tenet03, 
+            so the test fails because the L2VPN status is still up.
+        '''
+
+        ## Beginning of verification code
+        api_url_topology = SDX_CONTROLLER + '/topology'
+        response = requests.get(api_url_topology)
+        data = response.json()
+        links = {link["id"]: link for link in data["links"]}
+        link1 = "urn:sdx:link:tenet.ac.za:Tenet01/2_Tenet03/2"
+        assert links[link1]["status"] == "down", str(links[link1])
+        ## End of verification code
+
+        data = requests.get(api_url).json()
+        key = next(iter(data))
+        ## Beginning of verification code
+        path = [item['port_id'] for item in data[key]['current_path']]
+        assert path == ['urn:sdx:port:tenet.ac.za:Tenet01:50', 'urn:sdx:port:tenet.ac.za:Tenet03:50']
+        ## End of verification code
+        assert data[key]["status"] == "down"
+
+        # set one link to up
+        self.net.net.configLinkStatus('Tenet01', 'Tenet03', 'up')
+
+        time.sleep(15)
+
+        data = requests.get(api_url).json()
+        key = next(iter(data))
+        assert data[key]["status"] == "up"
+
+        # test connectivity
+        assert ', 0% packet loss,' in h6.cmd('ping -c4 10.1.1.8')
+
+    def test_011_send_topo_update_with_intra_domain_link_down_path_not_found(self):
```

Comment:
Here, the name of the method is inverted (as I mentioned before). A L2VPN between Tenet01 and Tenet02 will find alternative paths on inter-domain links.